crates/runtime/src/component/mod.rs

```diff
@@ -448,12 +346,52 @@ where
             // Don't link builtin instances or feature-gated instances if the feature is disabled
-            skip_static_instances!(name);
-            if rt.skip_feature_gated_instance(name) {
-                continue;
-            }
-            link_item(&engine, &mut linker.root(), [], ty, "", name, None)
-                .context("failed to link item")?;
+            match name.split_once('/').map(|(pkg, suffix)| {
+                suffix
+                    .split_once('@')
+                    .map_or((pkg, name, None), |(iface, version)| {
+                        (pkg, iface, Some(version))
+                    })
+            }) {
+                Some(
+                    ("wasi:blobstore", "blobstore" | "container" | "types", Some("0.2.0-draft"))
+                    | ("wasi:config", "runtime" | "store", Some("0.2.0-draft"))
+                    | ("wasi:keyvalue", "atomics" | "batch" | "store", Some("0.2.0-draft"))
+                    | ("wasi:logging", "logging", None | Some("0.1.0-draft"))
+                    | ("wasmcloud:bus", "lattice", Some("1.0.0" | "2.0.0"))
+                    | ("wasmcloud:messaging", "consumer" | "types", Some("0.2.0"))
+                    | ("wasmcloud:secrets", "reveal" | "store", Some("0.1.0-draft")),
+                ) => {}
+                Some((
+                    "wasi:cli",
+                    "environment" | "exit" | "stderr" | "stdin" | "stdout" | "terminal-input"
+                    | "terminal-output" | "terminal-stderr" | "terminal-stdin" | "terminal-stdout",
+                    Some(version),
+                )) if is_0_2(version, 0) => {}
+                Some(("wasi:clocks", "monotonic-clock" | "wall-clock", Some(version)))
+                    if is_0_2(version, 0) => {}
+                Some(("wasi:clocks", "timezone", Some(version))) if is_0_2(version, 1) => {}
+                Some(("wasi:filesystem", "preopens" | "types", Some(version)))
+                    if is_0_2(version, 0) => {}
+                Some((
+                    "wasi:http",
+                    "incoming-handler" | "outgoing-handler" | "types",
+                    Some(version),
+                )) if is_0_2(version, 0) => {}
+                Some(("wasi:io", "error" | "poll" | "streams", Some(version)))
+                    if is_0_2(version, 0) => {}
+                Some(("wasi:random", "insecure-seed" | "insecure" | "random", Some(version)))
+                    if is_0_2(version, 0) => {}
+                Some((
+                    "wasi:sockets",
+                    "instance-network" | "ip-name-lookup" | "network" | "tcp-create-socket" | "tcp"
+                    | "udp-create-socket" | "udp",
+                    Some(version),
+                )) if is_0_2(version, 0) => {}
+                _ if rt.skip_feature_gated_instance(name) => {}
+                _ => link_item(&engine, &mut linker.root(), [], ty, "", name, None)
+                    .context("failed to link item")?,
```

Comment:
```suggestion
                    .context("failed to link item")?,
```

More information about what failed to link would be great here via `with_context(...)`, but not blocking on that.